Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableSystemComponent.h

```diff
@@ -90,5 +102,11 @@ namespace AzFramework
 
         AZ::Data::AssetId m_rootSpawnableId;
         AZ::SettingsRegistryInterface::NotifyEventHandler m_criticalAssetsHandler;
+
+        // Cache of EntitySpawnTickets created from script, they are tracked to avoid
+        // refcounting issues at runtime
+        AZStd::unordered_set<EntitySpawnTicket> m_spawnedEntityTickets;
```

Comment:
Spawnables are created by many systems like game mode, networking, etc. So keeping this set in the system component to just track tickets created by scripts isn't a good idea imo.

I suggest putting this set in the SpawnableScriptMediator class itself. So, for each nodeable that defines a mediator, it'll maintain a set of tickets to avoid lifecycle issues. If you do that, you wouldn't even need a bus.

So, if the nodeable gets destroyed/deactivated, tickets will be cleared and things will get destroyed unless the user stores the tickets elsewhere in variables, which sounds like a correct workflow to me.

Reply:
Good suggestion! I'll try it and update after